anyway/models.py

```diff
@@ -871,13 +872,52 @@ class NewsFlash(Base):
     street2_hebrew = Column(Text(), nullable=True)
     non_urban_intersection_hebrew = Column(Text(), nullable=True)
     road_segment_name = Column(Text(), nullable=True)
+    critical = Column(Boolean(), nullable=True)
     newsflash_location_qualification = Column(
         Integer(),
         nullable=False,
         server_default=text(f"{NewsflashLocationQualification.NOT_VERIFIED.value}"),
     )
     location_qualifying_user = Column(BigInteger(), nullable=True)
 
+    def __init__(self, **kwargs):
+        super(NewsFlash, self).__init__(**kwargs)
+        self.critical = self.get_critical()
+
+    def get_critical(
+        self,
+        years_before=5,
+        suburban_road_severe_value=10,
+        suburban_road_killed_value=3,
+        urban_severe_value=2,
+    ):
+        from anyway.widgets.suburban_widgets.injured_count_by_severity_widget import (
+            InjuredCountBySeverityWidget,
+        )
+
+        if self.road1 is None or self.road_segment_name is None or self.date is None:
+            return None
+        resolution = BE_CONST.ResolutionCategories(self.resolution)
+        five_years_ago = self.date.replace(year=self.date.year - years_before)
+        critical_values = InjuredCountBySeverityWidget.get_injured_count_by_severity(
+            self.road1, self.road_segment_name, five_years_ago, self.date
+        )
+        critical = False
+        if (
+            resolution == BE_CONST.ResolutionCategories.SUBURBAN_ROAD
+            or resolution == BE_CONST.ResolutionCategories.URBAN_JUNCTION
```

Comment:
Relevant urban resolution is STREET

Reply:
just to make sure i understand , there is two options:
SUBURBAN_ROAD
STREET
?

Reply:
you are correct

Reply:
done